homeassistant/components/knx/__init__.py

```diff
@@ -336,9 +335,17 @@ async def service_send_to_knx_bus(self, call):
         """Service for sending an arbitrary KNX message to the KNX bus."""
         attr_payload = call.data.get(SERVICE_KNX_ATTR_PAYLOAD)
         attr_address = call.data.get(SERVICE_KNX_ATTR_ADDRESS)
+        attr_type = call.data.get(SERVICE_KNX_ATTR_TYPE)
 
         def calculate_payload(attr_payload):
             """Calculate payload depending on type of attribute."""
+            if attr_type is not None:
+                transcoder = DPTBase.parse_transcoder(attr_type)
+                if transcoder is not None:
+                    return DPTArray(transcoder.to_knx(attr_payload))
+                else:
+                    # We can proceed with the fallback here.
+                    _LOGGER.error("Invalid type for knx.send service: %s", attr_type)
```

Comment:
Is it fine to have this log for the user to know that something was wrong? We can still send the raw payload, but I honestly doubt it makes sense. Is there an exception we can throw from home assistant that signals the user that he send wrong data?

Reply:
Errors that are raised within the service handler will show as notification to user. Eg raise a `TypeError` with an explanation in the exception argument.

Reply:
Or maybe `ValueError` is more appropriate?

Reply:
I'll make a `ValueError`. Thanks!